server/command.go

```diff
@@ -43,7 +46,140 @@ func (p *Plugin) hasSysadminRole(userId string) (bool, error) {
+func validateCommand(action string, parameters []string) *model.CommandResponse {
+	switch action {
+	case "preview":
+		if len(parameters) != 1 {
+			return responsef("Please specify a team, for which preview should be made.")
+		}
+	case "list":
+		if len(parameters) > 0 {
+			return responsef("List command does not accept any extra parameters")
+		}
+	case "set_channel_welcome":
+		if len(parameters) == 0 {
+			return responsef("`set_channel_welcome` command requires the message to be provided")
+		}
+	case "get_channel_welcome":
+		if len(parameters) > 0 {
+			return responsef("`get_channel_welcome` command does not accept any extra parameters")
+		}
+	case "delete_channel_welcome":
+		if len(parameters) > 0 {
+			return responsef("`delete_channel_welcome` command does not accept any extra parameters")
+		}
+	}
+
+	return nil
+}
+
+func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) *model.CommandResponse {
+	found := false
+	for _, message := range p.getWelcomeMessages() {
+		if message.TeamName == teamName {
+			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
+				return responsef("error occured while processing greeting for team `%s`: `%s`", teamName, err)
+			}
+
+			found = true
+		}
+	}
+
+	if !found {
+		return responsef("team `%s` has not been found", teamName)
+	}
+
+	return &model.CommandResponse{}
+}
+
+func (p *Plugin) executeCommandList(args *model.CommandArgs) *model.CommandResponse {
+	wecomeMessages := p.getWelcomeMessages()
+
+	if len(wecomeMessages) == 0 {
+		return responsef("There are no welcome messages defined")
+	}
+
+	// Deduplicate entries
+	teams := make(map[string]struct{})
+	for _, message := range wecomeMessages {
+		teams[message.TeamName] = struct{}{}
+	}
+
+	var str strings.Builder
+	str.WriteString("Teams for which welcome messages are defined:")
+	for team := range teams {
+		str.WriteString(fmt.Sprintf("\n * %s", team))
+	}
+	return responsef(str.String())
+}
+
+func (p *Plugin) executeCommandSetWelcome(args *model.CommandArgs) *model.CommandResponse {
+	channelInfo, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return responsef(
+			"error occured while checking the type of the chanelId `%s`: `%s`",
+			args.ChannelId, appErr)
+	}
+
+	if channelInfo.Type == model.CHANNEL_PRIVATE {
+		return responsef("welcome messages are not supported for direct channels")
+	}
+
+	// strings.Fields will consume ALL whitespace, so plain re-joining of the
+	// parameters slice will not produce the same message
+	message := strings.SplitN(args.Command, "set_channel_welcome", 2)[1]
+	message = strings.TrimSpace(message)
+
+	key := fmt.Sprintf("%s%s", args.ChannelId, WELCOMEBOT_CHANNEL_WELCOME_KEY)
+	if appErr := p.API.KVSet(key, []byte(message)); appErr != nil {
+		return responsef(
+			"error occured while storing the welcome message for the chanel: `%s`",
+			appErr)
+	}
```

Comment:
```suggestion
		return responsef("error occured while storing the welcome message for the chanel: `%s`", appErr)
	}
```

Reply:
Most code in MM will have this as a one-liner.  Let's use the one-liner for consistency.